challenge-16/challenge-16.js

```diff
@@ -1,70 +1,106 @@
-- "e é a 2ª letra do meu nome."
-E assim por diante, até a última.
-*/
-console.log( 'As letras do seu nome:' );
-// ?
+  /*
+  Declare uma variável chamada `name` que receba seu primeiro nome.
+  Mostre no console todas as letras do seu nome separadas, com a frase:
+  - "[LETRA] é a [POSIÇÃO]ª letra do meu nome."
+  Ex: no caso do nome ser "Fernando", deve mostrar as frases:
+  - "F é a 1ª letra do meu nome."
+  - "e é a 2ª letra do meu nome."
+  E assim por diante, até a última.
+  */
+  console.log( 'As letras do seu nome:' );
+  var name = 'Natan';
 
-/*
-- Declare uma variável chamada `fullName`, que receba seu nome completo,
-escrito no formato de slug (caixa baixa e palavras separadas por um traço).
-Ex: o nome "Fernando Daciuk" ficaria "fernando-daciuk"
-- Faça a primeira letra de cada nome ficar em caixa alta, e troque o traço
-por um espaço.
-- Detalhe: o código que você escrever abaixo deve funcionar para qualquer
-nome, então fique à vontade para usar as artimanhas que já viu até agora no
-curso para fazer isso funcionar corretamente :)
-- Mostre no console o nome no formato slug, e o resultado final. Use um
-console.log para cada formato.
-*/
-console.log( '\nNome convertido à partir de um slug:' );
-// ?
+  for( var index = 0; index < name.length; index++ ) {
+    console.log( name[ index ] + ' é a ' + ( index + 1 ) + 'ª letra do meu nome.' );
+  }
 
-/*
-- Crie um array com 5 nomes. Reduza esses nomes a uma única string, separando
-cada nome por vírgula. Entre o penúltimo e o último nome, o separador deve
-ser um "e". No final, concatene com a frase: " são meus amigos.".
-O resultado final deve ficar mais ou menos assim:
-- "João, Maria, Roberto, Pedro e Marcos são meus amigos."
-- Detalhe: o código abaixo deve funcionar com um array de qualquer tamanho.
-5 nomes foi somente uma sugestão ;)
-*/
-console.log( '\nMeus amigos:' );
-// ?
+  /*
+  - Declare uma variável chamada `fullName`, que receba seu nome completo,
+  escrito no formato de slug (caixa baixa e palavras separadas por um traço).
+  Ex: o nome "Fernando Daciuk" ficaria "fernando-daciuk"
+  - Faça a primeira letra de cada nome ficar em caixa alta, e troque o traço
+  por um espaço.
+  - Detalhe: o código que você escrever abaixo deve funcionar para qualquer
+  nome, então fique à vontade para usar as artimanhas que já viu até agora no
+  curso para fazer isso funcionar corretamente :)
+  - Mostre no console o nome no formato slug, e o resultado final. Use um
+  console.log para cada formato.
+  */
+  console.log( '\nNome convertido à partir de um slug:' );
+  var fullName = 'natan-siqueira-alves';
+  var fullNameResult = '';
 
-/*
-Usando o replace(), faça a string "Roberto" virar "Roberta".
-Mostre o resultado no console.
-*/
-console.log( '\nEra "Roberto", agora é:' );
-// ?
+  for( var i = 0; i < fullName.split( '-' ).length; i++ ) {
+    fullNameResult += fullName.split( '-' )[ i ].charAt( 0 ).toUpperCase() + fullName.split( '-' )[ i ].substring(1) + ' ';
```

Comment:
-1
O seu exemplo funciona! Mas usando dessa forma, você estará "gastando" mais memória quando a sua aplicação for acessada. Por que?

Lembre-se que, a cada novo objeto criado, é separado um espaço na memória para esse objeto. Quando você o `split`, você não modifica o valor da variável `fullName`, mas cria um novo array.

Como você está fazendo `fullName.split( '-' )` 3 vezes, você está colocando o mesmo valor em 3 espaços diferentes da memória, sendo que você poderia utilizar somente 1, atribuindo esse array à uma variável :)

E já que você está utilizando um array, você consegue utilizar, por exemplo, o método `map` para gerar os nomes, ao invés de utilizar o `for`. 

São algumas variáveis a menos que você cria, e ainda deixa seu código mais legível :)